### PR TITLE
feat(iOS): added custom URL schemes handling in the `AppDelegate` class

### PR DESCRIPTION
**Pull request by CampioneDev**
Until now, only ["associated domains"](https://developer.apple.com/documentation/xcode/supporting-associated-domains) were handled, using the `application_continue` function, implementing [this Swift method from the `UIApplicationDelegate` class](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623072-application).

For [custom URL schemes](https://developer.apple.com/documentation/xcode/defining-a-custom-url-scheme-for-your-app), I added a new `application_open_url` function that implements the signature of [this other Swift method](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623112-application).

The code of the pre-existing `application_continue` function has been moved into a separate `handle_deep_link` function so the new `application_open_url` function can call it as well.  
For now, I believe it's appropriate to use the same `Event::Opened` event in both situations. Since the scheme is part of the URL, a listener can differentiate between them if needed.

**Tauri:** this change works automatically with the ["Deep Linking" plugin](https://v2.tauri.app/plugin/deep-linking/) without further modifications.  
Custom URL schemes in mobile apps are essential, for example, when dealing with OAuth redirect URLs.

### CODE REVIEW

**Review by pewsheen — APPROVED** [review]
@CampioneDev Thanks for the contribution! I have tested it, and it works on my side 👍 

Before we merge this PR, we'll need you to do some trivial things:
- Add a change log file: https://github.com/tauri-apps/tao/tree/dev/.changes
  - You can find a file in the `.changes` folder as a reference. I think using `patch` will be enough
- Sign your commit: https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits

**Comment by CampioneDev** [comment]
Thanks!

(BTW This is my first PR and contribution to OSS **ever**... I hope everything's ok, and sorry for the force-pushed mess)

File: .changes/ios-custom-url-schemes.md
Comment (pewsheen) [inline .changes/ios-custom-url-schemes.md]:
```suggestion
On iOS, implement `application:openURL:options:` to handle custom URL schemes.
```

This might be too detailed xD. The changelog will show on the release note with a link to the PR. I think leaving a brief message will be enough. Since English is not my native language, feel free to change the suggested changelog.

**Comment by FabianLars** [comment]
I also requested @lucasfernog's review because of his involvement in deep links and as the author of https://github.com/tauri-apps/tao/pull/752

**Comment by lucasfernog** [comment]
> Thanks!
> 
> (BTW This is my first PR and contribution to OSS **ever**... I hope everything's ok, and sorry for the force-pushed mess)

your first OSS contribution is a objc-in-rust PR? omg that's impressive

**Comment by lucasfernog** [comment]
my first contribution was a dumb issue like "hey you need to fix this typo" people almost deleted my GitHub account 😂 😂

**Review by lucasfernog — APPROVED** [review]
well done!

**Comment by lucasfernog** [comment]
I'll force merge this one but would be great if you could setup commit signing in the future (even for your own personal usage!)

**Pull request merged**
